File: src/diet/Omnivore.cpp
/* NIM/Nama  : 13515078 / Veren Iliana K   */
/* File      : Omnivore.cpp                */

#include "Omnivore.h"

Omnivore::Omnivore()
{
	meatRatio = 0.3;
	vegRatio = 0.3;
}

Omnivore::Omnivore(const Omnivore& O)
{
	meatRatio = O.meatRatio;
	vegRatio = O.vegRatio;
}

Omnivore& Omnivore::operator= (const Omnivore& O)
{
	meatRatio = O.meatRatio;
	vegRatio = O.vegRatio;
	return *this;
}

double Omnivore::getMeatRatio() const{
	return meatRatio;
}

double Omnivore::getVegRatio() const{
	return vegRatio;
}

File: virtual-zoo-java/src/animal/diet/Omnivore.java
/**
 * @author NIM/Nama : 13515078 / Veren Iliana Kurniadi
 * File : Omnivore.java
 */

package animal.diet;

public interface Omnivore {
  /**
   * Mengembalikan nilai rasio daging.
   * @return nilai rasio daging
   */
  public double getMeatRatio();

  /**
   * Mengembalikan nilai rasio sayur.
   * @return nilai rasio sayur
   */
  public double getVegRatio();
}
